src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -326,13 +326,26 @@ private ValueTask<HttpConnection> GetOrReserveHttp11ConnectionAsync(Cancellation
 
             // See if we have an HTTP2 connection
             Http2Connection http2Connection = _http2Connection;
+
+            if (http2Connection != null)
+            {
+                TimeSpan pooledConnectionLifetime = _poolManager.Settings._pooledConnectionLifetime;
+                // Check only lifetime expiration. There is no need to check idle if we are about to use it.
```

Comment:
I don't think we need this comment, it only makes sense if you know that we used to check idle for HTTP/1.1.